pandas/core/frame.py

```diff
@@ -3137,8 +3137,13 @@ def query(self, expr, inplace=False, **kwargs):
 
             .. versionadded:: 0.25.0
 
-            You can refer to column names that contain spaces by surrounding
-            them in backticks.
+            You can refer to column names that contain spaces or operators by
+            surrounding them in backticks. This way you can also escape
+            names that start with a digit, or are a Python keyword. Basically
+            when it is not valid Python identifier.
+
+            NB. You cannot use this if there are multiple invalid characters
```

Comment:
Is there a practical reason for this limitation? Seems a little arbitrary

Reply:
Good catch, this was sloppy. This is not true.

However, there is an edge case here, it is a bit more complicated. I could replace it too: 

            NB. A combination of an invalid character with a space, or
            multiple spaces will collapse the space. The reason is that the
            string will be parsed as Python code
            and in Python ``a*b`` and ``a* b`` are equal.
            So, if you want to refer to a column called ``"a* b"`` you
            can use ```a*b```, but it is recommend in this case to change the
            column name or use bracket notation, to prevent very opaque code.

But, since this is such an edge case, it would only make the documentation confusing. Just leave it out?

Reply:
So IIUC here the danger is that valid expressions will be evaluated? Or is this a matter of how we are handling whitespace in this PR?

Reply:
No, it isn't a matter of expressions being evaluated, it is a matter of expressions being parsed, and they are parsed as Python code, and therefore whitespace sometimes disappears.

Let's just say, there are some edge cases (which cannot be solved easily), but they are rare. Should we still mention them somewhere?

Reply:
Can you see if it can be merged now?

Reply:
This edge case is now largely resolved. (Although, one can still construct certain query + column names combination to confuse the parser, but such cases are much rarer now.)

Reply:
I could maybe see if I can make code that detects such edge cases and raises an appropriate exception. (Now they will go by silent, and the result may be false or correct.)

Reply:
Done.